### PR TITLE
Add feature doc template

### PR DESCRIPTION
**Pull request by tssurya**
#### Which issue(s) this PR fixes

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: docs/features/template.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+Explain how the user will use the feature. Be detailed and explicit.
+Describe all of the actors, their roles, and the APIs or interfaces
+involved. Define a starting state and then list the steps that the
+user would need to go through to trigger the feature described in the
+enhancement. **A detailed architectural diagram is a must to showcase
+how this feature works in our CNI.** Remember that a picture can speak
+a thousand words.
+
+
+### Implementation Details
```
Comment (l8huang) [inline docs/features/template.md:74]:
```suggestion
## Implementation Details
```

File: docs/features/template.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
+# Name of the Feature
+
+* Add diagrams wherever possible to explain things better
+* Keep the end user in mind as target audience when you write thisdocumentation; not an ovn expert
```
Comment (kyrtapz) [inline docs/features/template.md:4]:
```suggestion
* Keep the end user in mind as target audience when you write this documentation; not an ovn expert
```

File: docs/features/template.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+For Example: The admin wishes to protect a sensitive namespace by applying an
+AdminNetworkPolicy which denies ingress from all other in-cluster resources
+for all ports and protocols.
+
+## How to enable this feature on an OVN Kubernetes cluster?
+
+Did you write new configs or knobs that need to be turned on
+for this feature? If so educate end users on those details.
+How can users disabled this feature if they don't want it?
```
Comment (kyrtapz) [inline docs/features/template.md:61]:
```suggestion
How can users disable this feature if they don't want it?
```

File: docs/features/template.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+### User facing API Changes
+
+If any API changes were done, they must be outlined here and
+look at the developer docs on how to generate API reference
+documentation for this feature. Provide a link to that API
+reference web page from this spot.
+Also add details on where the CRD lives; example is it in
+kubernetes or kubernetes-sigs or network-plumbing-working-group?
+
+### OVN Kubernetes Implementation Details
```
Comment (kyrtapz) [inline docs/features/template.md:87]: Should it be `OVN-Kubernetes` instead?

**Review by kyrtapz — COMMENTED** [review]
Overall looks good to me!
Do you think it  would be worth to add a section about dependencies and something about whether the feature in question affects other features?

File: docs/features/template.md (hunk 4 above)
Comment (girishmg) [reply docs/features/template.md:87]: yes, using OVN-Kubernetes, will be better.

Comment (tssurya) [reply docs/features/template.md:87]: hmm the recommendation I got from PR team/docs team when I wrote https://www.redhat.com/en/blog/how-the-new-ovn-kubernetes-architecture-helps-with-scaling-openshift was that we must call it either as `ovn-kubernetes` repo name OR `OVN Kubernetes`; hyphenating is not correct, we can make it `OVNKubernetes` as well ?

File: docs/features/template.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+Provide details on specific changes done to br-ex, br-int or how
+the relevant openflows generated by this feature look like.
+Keep in mind goal is to educate users how this
+feature was implemented using OVS in this section.
+
+## Future Items
+
+Add a bullet list of future plans if any
+
```
Comment (tssurya) [inline docs/features/template.md:121]: add troubleshooting section

Comment (tssurya) [reply docs/features/template.md:121]: also best practices and scale information

File: docs/features/template.md (hunk 4 above)
Comment (tssurya) [reply docs/features/template.md:87]: going with OVN-Kubernetes, resolving issue

File: docs/features/template.md (hunk 1 above)
Comment (tssurya) [reply docs/features/template.md:74]: done.

File: docs/features/template.md (hunk 3 above)
Comment (tssurya) [reply docs/features/template.md:61]: done

File: docs/features/template.md (hunk 2 above)
Comment (tssurya) [reply docs/features/template.md:4]: done

**Review by kyrtapz — APPROVED** [review]

**Pull request merged**